docs/public/specifications/tests/Nameserver-TP/nameserver15.md

```diff
@@ -0,0 +1,119 @@
+# NAMESERVER15: Checking for software version
+
+## Test case identifier
+
+**NAMESERVER15**
+
+## Table of contents
+
+* [Objective](#objective)
+* [Scope](#scope)
+* [Inputs](#inputs)
+* [Summary](#summary)
+* [Test procedure](#test-procedure)
+* [Outcome(s)](#outcomes)
+* [Special procedural requirements](#special-procedural-requirements)
+* [Intercase dependencies](#intercase-dependencies)
+* [Terminology](#terminology)
+
+## Objective
+
+This Test Case verifies if a name server responds to TXT queries in the CHAOS class, specifically
+about its software version as it may sometimes be desirable not to reveal that information.
+
+A description of DNS classes can be found in [RFC2929], section 3.2.
+
+## Scope
+
+It is assumed that *Child Zone* is also tested and reported by [Connectivity01].
+This Test Case will just ignore non-responsive name servers or name servers not
+giving a correct DNS response for an authoritative name server.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+Message Tag                 | Level   | Arguments                     | Message ID for message tag
+:---------------------------|:--------|:------------------------------|:---------------------------------------------------------------------------------------
+N15_BIND_VERSION            | INFO    | ns_ip_list, string            | Name server(s) "{ns_ip_list}" respond(s) to CHAOS TXT query "version.bind" with value "{string}".
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine Profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [Argument List].
+
+## Test procedure
+
+1. Create the following empty sets:
+   1. Name server IP and string ("TXT Data")
+
+2. Create [DNS Query] with query type TXT, query class CHAOS and
+   query name "version.bind" ("TXT Query").
+
+3. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+4. For each name server in *Name Server IP* do:
+   1. [Send] *TXT Query* to the name server and collect the response.
+   2. If at least one of the following criteria is met, go to next name server:
+      1. There is no DNS response.
+      2. The [DNS Response] does not have the [RCODE Name] NoError.
+      3. The [DNS Response] does not have any TXT record in the
+         answer section with owner name "version.bind".
+   3. Extract the string from the RDATA of the TXT record in the
+      answer section of the [DNS Response].
```

Comment:
There should be a definition of what “extracting the string” means, because a TXT resource record can contain one or more strings. I suggest that each string be concatenated into one large string without addition of whitespace, as if by `join("", @strings)`, like what is done for SPF and DMARC.

Reply:
Good point. Should that process be added to "Default handling of a DNS Response" in [DNS Query and Response Defaults] so that we do not have to repeat that every time a TXT record is processed?

[DNS Query and Response Defaults]: https://github.com/zonemaster/zonemaster/blob/develop/docs/public/specifications/tests/DNSQueryAndResponseDefaults.md